File: src/db/schema/printer.ts
import { sqliteTable, text, integer } from "drizzle-orm/sqlite-core";

export const printers = sqliteTable("printers", {
  id: text("id").primaryKey().notNull(),
  name: text("name").notNull(),
  printer_type: text("printer_type").notNull(), // "network", "usb", "bluetooth"
  ip_address: text("ip_address"),
  port: integer("port"),
  is_active: integer("is_active").default(0).notNull(),
  created_at: text("created_at"),
  updated_at: text("updated_at"),
});

File: src-tauri/src/db/models/printer.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Printer {
    pub id: String,
    pub name: String,
    pub printer_type: String, // "network", "usb", "bluetooth"
    pub ip_address: Option<String>,
    pub port: Option<i32>,
    pub is_active: bool,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}
